core_api/src/build_chains.py

```diff
@@ -62,82 +71,76 @@ async def build_vanilla_chain(
-    question = chat_request.message_history[-1].text
-    previous_history = list(chat_request.message_history[:-1])
-
-    chain = make_stuff_document_runnable(system_prompt=summarisation_prompt, llm=llm)
-
-    documents: list[Chunk] = []
-    for selected_file in chat_request.selected_files:
-        chunks = get_file_chunked_to_tokens(
-            file_uuid=selected_file.uuid,
-            user_uuid=user_uuid,
-            storage_handler=storage_handler,
-        )
-        documents += chunks
-
-    # right now, can only handle a single document so we manually truncate
-    max_tokens = 20_000  # parameterise later
-    doc_token_sum = np.cumsum([doc.token_count for doc in documents])
-    doc_token_sum_limit_index = len([i for i in doc_token_sum if i < max_tokens])
+    @chain
+    def make_document_context(input_dict):
+        documents: list[Chunk] = []
+        for selected_file in input_dict["file_uuids"]:
+            chunks = get_file_chunked_to_tokens(
+                file_uuid=selected_file,
+                user_uuid=input_dict["user_uuid"],
+                storage_handler=storage_handler,
+            )
+            documents += chunks
+
+        # right now, can only handle a single document so we manually truncate
+        max_tokens = 20_000  # parameterise later
+        doc_token_sum = np.cumsum([doc.token_count for doc in documents])
+        doc_token_sum_limit_index = len([i for i in doc_token_sum if i < max_tokens])
+
+        documents_trunc = documents[:doc_token_sum_limit_index]
+        if len(documents) < doc_token_sum_limit_index:
+            log.info("Documents were longer than 20k tokens. Truncating to the first 20k.")
+        return documents_trunc
+
+    return (
+        RunnablePassthrough.assign(documents=(make_document_context | RunnableLambda(format_chunks)))
+        | make_chat_prompt_from_messages_runnable(system_prompt, question_prompt)
+        | llm
+        | {"response": StrOutputParser()}
+    )
 
-    documents_trunc = documents[:doc_token_sum_limit_index]
-    if len(documents) < doc_token_sum_limit_index:
-        log.info("Documents were longer than 20k tokens. Truncating to the first 20k.")
 
-    params = {
-        "question": question,
-        "documents": documents_trunc,
-        "messages": [(msg.role, msg.text) for msg in previous_history],
+def get_routable_chains(
+    retrieval_chain: Annotated[Runnable, Depends(build_retrieval_chain)],
+    summary_chain: Annotated[Runnable, Depends(build_summary_chain)],
+):
+    return {
+        "default": retrieval_chain,
+        "info": ChatPromptTemplate.from_template(INFO_RESPONSE),
+        "ability": ChatPromptTemplate.from_template(ABILITY_RESPONSE),
+        "coach": ChatPromptTemplate.from_template(COACH_RESPONSE),
+        "gratitude": ChatPromptTemplate.from_template("You're welcome!"),
+        "retrieval": retrieval_chain,
+        "summarisation": summary_chain,
+        "extract": ChatPromptTemplate.from_template(
```

Comment:
```suggestion
        "extract": ChatPromptTemplate.from_template(
```
out of scope i know but lets kill this one off